File: src/main/java/br/com/belezaNaWeb/javaTest/service/ProductService.java
package br.com.belezaNaWeb.javaTest.service;

import javax.annotation.PostConstruct;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.belezaNaWeb.javaTest.domain.Product;
import br.com.belezaNaWeb.javaTest.repository.ProductRepository;

@Service
public class ProductService {

	@Autowired
	private ProductRepository productRepository;

	@PostConstruct
	public void init() {
		productRepository = new ProductRepository();
	}

	public Product findProductBySku(Long sku) {
		return productRepository.findBySku(sku);
	}

	public void createProduct(Product product) throws Exception {
		if (product == null) {
			throw new Exception("The Product can't be null.");
		}

		Product productExist = productRepository.findBySku(product.getSku());

		if (productExist != null) {
			throw new Exception("The Product with sku " + product.getSku() + " already exists.");
		} else {
			productRepository.create(product);
		}
	}

	public void deleteProduct(Long sku) {
		Product product = productRepository.findBySku(sku);
		if (product != null) {
			productRepository.delete(product);
		}
	}

	public boolean updateProduct(Product product) {
		Product productNew = productRepository.findBySku(product.getSku());
		if (productNew != null) {
			productNew.setName(product.getName());
			productNew.setInventory(product.getInventory());
			productRepository.update(productNew);
		}
		return productNew != null ? true : false;
	}

}

File: src/main/kotlin/br/com/blz/testjava/application/service/ProductService.kt
package br.com.blz.testjava.application.service

import br.com.blz.testjava.domain.entity.Product
import br.com.blz.testjava.domain.repository.ProductRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ProductService @Autowired constructor(
    private val repository: ProductRepository
){

    fun save(product: Product): Product {
        return repository.save(product)
    }

    fun getAll(): MutableList<Product> {
        return repository.getAll()
    }

    fun get(sku: Int): Product {
        return repository.get(sku)
    }

}
